adapter/memQuota/memQuota.go

```diff
@@ -296,27 +288,25 @@ func (mq *memQuota) reapDedup() {
 	}
 }
 
-// Produce a unique key representing the cell
-func makeKey(args adapter.QuotaArgs) string {
+// Produce a unique key representing the given labels.
+func makeKey(name string, labels map[string]interface{}) string {
 	ws := keyWorkspacePool.Get().(*keyWorkspace)
 	keys := ws.keys
 	buffer := ws.buffer
 
-	// get all the label names
-	for k := range args.Labels {
+	// ensure stable order
+	for k := range labels {
 		keys = append(keys, k)
 	}
-
-	// ensure stable order
 	sort.Strings(keys)
 
-	buffer = append(buffer, []byte(args.Name)...)
+	buffer = append(buffer, []byte(name)...)
```

Comment:
question (curious): why are we not using `bytes.Buffer` for the buffer here?

It has a `Reset()` that retains the underlying storage, and has a `WriteString` method that seems more natural here.